Applied clang-tidy readability-non-const-parameter warning fixes auto…

seanm opened the pull request:
…matically

seanm commented:
@qkoziol perhaps this is a step forward in the const battle.  This patch was computer generated by running:

`run-clang-tidy.py -checks=-*,readability-non-const-parameter -fix`

The tool seems smart enough to change only what results in something that compiles. :)

seanm committed:
Automatically applied clang-tidy readability-avoid-const-params-in-decls fixes

Removes useless const declarations.

byrnHDF reviewed (approved):
Assuming source formatting fixed.

gnuoyd commented:
This looks like a step in the right direction.

qkoziol commented on test/direct_chunk.c:
```diff
@@ -76,9 +76,9 @@
+                            size_t nbytes, const size_t *buf_size, void **buf);
 static size_t filter_bogus2(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                            size_t nbytes, size_t *buf_size, void **buf);
+                            size_t nbytes, const size_t *buf_size, void **buf);
```
These 2 may not work, since they are supposed to match the signature for H5Z_func_t

qkoziol commented on test/dsets.c:
```diff
@@ -307,7 +307,7 @@ static size_t filter_bogus(unsigned int flags, size_t cd_nelmts, const unsigned
 static herr_t set_local_bogus2(hid_t dcpl_id, hid_t type_id, hid_t space_id);
 static size_t filter_bogus2(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                            size_t nbytes, size_t *buf_size, void **buf);
+                            size_t nbytes, const size_t *buf_size, void **buf);
```
Same here (H5Z_func_t)

qkoziol commented on tools/test/h5dump/h5dumpgentest.c:
```diff
@@ -150,7 +150,7 @@ const H5Z_class2_t H5Z_MYFILTER[1] = {{
 
 static size_t H5Z_filter_dynlibud(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                                  size_t nbytes, size_t *buf_size, void **buf);
+                                  size_t nbytes, const size_t *buf_size, void **buf);
```
Same here (H5Z_func_t)

qkoziol reviewed (approved):
Aside from the H5Z_func_t issues, this looks good.

seanm commented:
@qkoziol good catch. see 3rd commit for followup to automated changes.

qkoziol commented:
> @qkoziol good catch. see 3rd commit for followup to automated changes.

The compiler is OK with this?   The H5Z_class2_t's 'filter' function pointer will be off slightly, since the buf_size pointer is const for these functions, but the H5Z_func_t typedef is non-const.

seanm commented:
I changed the `H5Z_func_t` typedef to const too.

It compiles for me.

qkoziol commented:
> I changed the `H5Z_func_t` typedef to const too.
> 
> It compiles for me.

Ah, missed that.  Unfortunately, that typedef can't change, since it's part of the public API for I/O filters.

seanm commented:
I'm not looking to change this in 1.10 or other stable branches.  Can there never ever be public API changes ever?  :(  This is about as minor an API change as you can get too.

qkoziol commented:
> I'm not looking to change this in 1.10 or other stable branches. Can there never ever be public API changes ever? :( This is about as minor an API change as you can get too.

Sure the public API can be changed.  :-)   In this case though, we shouldn't change it, since the buf_size pointer is an in/out value.   The filter will update it with the new size of the buffer, after it's been compressed (or otherwise filtered).

seanm commented:
Oh!  Odd that it compiles then.

seanm committed:
Fixed most readability-non-const-parameter warnings

These changes were made automatically by clang-tidy, but I manually reverted the changes related to the H5Z_func_t signature.

seanm commented:
@qkoziol ok, I think I fixed all the `H5Z_func_t` cases.

lrknox committed:
Reformat source with clang v10.0.1.
